modules/runner/src/integrationTest/java/org/apache/ignite/internal/table/ItTableScanTest.java

```diff
@@ -336,6 +337,36 @@ public void testGetAndDeleteDuringPureTableScan() throws Exception {
         );
     }
 
+    /**
+     * Ensures that multiple consecutive scan requests with different requested rows amount
+     * return the expected total number of requested rows.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testCompositeScanRequest() throws Exception {
+        int[][] demandPairs = {{3, 1}, {1, 3}};
+
+        for (int[] input : demandPairs) {
+            List<BinaryRow> scannedRows = new ArrayList<>();
+            Publisher<BinaryRow> publisher = internalTable.scan(0, null, null, null, null, 0, null);
+            CompletableFuture<Void> scanned = new CompletableFuture<>();
+
+            Subscription subscription = subscribeToPublisher(scannedRows, publisher, scanned);
+
+            subscription.request(input[0]);
+            subscription.request(input[1]);
+
+            int total = input[0] + input[1];
+            waitForCondition(() -> scannedRows.size() == total, 10_000);
+            assertEquals(total, scannedRows.size(), "input=" + Arrays.toString(input));
+
+            // Close the publisher.
+            subscription.request(1_000);
```

Comment:
why do we need such request ? can we just cancel it here ?

Reply:
Done, thanks.